### PR TITLE
feat: build admin dashboard (#28)

### LINKED ISSUE #28: Build FuzzyCat admin dashboard (metrics, clinic management, payments, risk)
### What
Build the internal FuzzyCat admin portal for monitoring the platform.

### Tasks
1. **Dashboard** — `app/admin/dashboard/page.tsx`:
   - Platform-wide metrics: total enrollments, active plans, revenue, default rate
   - Daily/weekly/monthly trends
   - Risk pool health indicator
2. **Clinics** — `app/admin/clinics/page.tsx`:
   - All registered clinics (status, enrollment count, revenue generated)
   - Approve/suspend clinics
   - View Stripe Connect status
3. **Payments** — `app/admin/payments/page.tsx`:
   - All payments across platform (filterable by status, clinic, date)
   - Failed payments requiring attention
   - Retry management
4. **Risk** — `app/admin/risk/page.tsx`:
   - Risk pool balance and health
   - Default rate trends
   - Defaulted plans list
   - Guarantee claims history
5. All routes protected (admin role only)

### Acceptance Criteria
- [ ] Admin sees platform-wide metrics
- [ ] Can manage clinics (approve/suspend)
- [ ] Can view and manage all payments
- [ ] Risk pool dashboard shows health metrics
- [ ] Protected by admin role

### PR DESCRIPTION
**Pull request by ynvtlmr**
## Summary
- **Admin tRPC router**: Added 8 new procedures (`getPlatformStats`, `getClinics`, `updateClinicStatus`, `getPayments`, `retryPayment`, `getRiskPoolDetails`, `getDefaultedPlans`, `getRecentAuditLog`) with admin-only access control and audit logging on mutations
- **Admin frontend**: Built 4 pages (Dashboard, Clinics, Payments, Risk Pool) with sidebar navigation, following the established clinic dashboard patterns (shadcn/ui components, tRPC hooks, status badges, skeleton loading states, pagination)
- **Unit tests**: 20 tests covering all new router procedures (success cases, empty/edge cases, mutation behavior)

## Test plan
- [x] `bun test server/__tests__/admin-router.test.ts` — 20 tests pass
- [x] `bun run check` — Biome lint/format passes
- [x] `bun run typecheck` — TypeScript type check passes
- [x] `bun test lib server app` — Full test suite passes (413 pass; 2 pre-existing plaid test isolation failures unrelated to this PR)
- [x] Pre-commit hooks pass (secrets, biome, typecheck, commitlint)
- [x] Pre-push hooks pass (circular dependency check, security scan, production build)
- [ ] Manual testing: verify admin pages render correctly when logged in as admin user

Closes #28

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/admin/payments/_components/payment-list.tsx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  const [dateFrom, setDateFrom] = useState('');
+  const [dateTo, setDateTo] = useState('');
+  const [offset, setOffset] = useState(0);
+  const limit = 20;
+
+  const { data, isLoading, error } = useQuery(
+    trpc.admin.getPayments.queryOptions({
+      status: statusFilter === 'all' ? undefined : statusFilter,
+      dateFrom: dateFrom ? new Date(dateFrom).toISOString() : undefined,
+      dateTo: dateTo ? new Date(dateTo).toISOString() : undefined,
```
[the remark this answers is not in the document]
Comment (ynvtlmr) [reply app/admin/payments/_components/payment-list.tsx]: FIXED: Explicitly constructed UTC date strings for date range filtering to ensure consistent behavior across timezones.
